internal/service/kafka.go

```diff
@@ -60,6 +64,10 @@ func (container *Container) decorateKafkaClientWithOpenTelemetry(producer *kafka
 	)
 }
 
+func (container *Container) decorateKafkaClientWithDebugger(client kafka.Client) kafka.Client {
```

Comment:
It think you can only decorate the client if `container.Cfg.HttpServer.DebugEnabled == true` and always apply the decorator to the client.